### ISSUE TITLE
Wrangler shows ugly output for wrangler preview with workers that return binary data

### ISSUE THREAD

**Issue by xortive** [cloudflare/wrangler#218]
running wrangler preview outputs the result of a GET request to your worker to stdout. For workers that return binary data like images, this is rather ugly and fills your console with special characters, is this something that should be fixed?

**Comment by xtuc** [comment]
I'm not sure how the preview works but we could check the Content-Type before showing it to the user.

**Comment by ashleygwilliams** [comment]
this is the type of thing that's right on the line of a bug and a feature! currently this is working as designed, it returns the exact thing the worker returns. this is not always useful tho! which means we should improve the design of this! i like the idea of the content-type, but i'm not sure that just removing things from the output is a good idea. 

let's brainstorm in this thread!

**Comment by xtuc** [comment]
> but i'm not sure that just removing things from the output is a good idea.

Can you please clarify? We would remove the non printable responses.

**Comment by pwfff** [comment]
Piggy-backing on this instead of making a new issue:
It would be nice to provide options on how to render the output. In addition to content-type detection to not throw junk on the terminal, when I *want* to see the junk, I often want to pipe it to something else. This doesn't work that well currently though:
```
$ wrangler preview | xxd | head
00000000: e2ac 87ef b88f 2049 6e73 7461 6c6c 696e  ...... Installin
00000010: 6720 7761 736d 2d70 6163 6b2e 2e2e 0ae2  g wasm-pack.....
00000020: ac87 efb8 8f20 496e 7374 616c 6c69 6e67  ..... Installing
00000030: 2077 7261 6e67 6c65 726a 732e 2e2e 0ae2   wranglerjs.....
00000040: 9ca8 2020 2042 7569 6c74 2073 7563 6365  ..   Built succe
```

**Comment by gabbifish** [comment]
It's been a while since this has last been picked up, and this is some good low-hanging fruit for us to take on! I still think we should absolutely provide an option for Wrangler to print ugly binary text to stdout, and I think it can be based on curl's approach:

```console
$ curl https://upload.wikimedia.org/wikipedia/commons/thumb/9/9d/Britishblue.jpg/800px-Britishblue.jpg
Warning: Binary output can mess up your terminal. Use "--output -" to tell curl to output it to your terminal anyway, or consider "--output <FILE>" to save to a file.
```

We can provide an error message like this and support identical flags. We can use the same heuristic as curl to determine whether a response is binary output or not: https://github.com/curl/curl/commit/5385450afd61328e7d24b50eeffc2b1571cd9e2f

This also brings us to @pwfff's point: it would be really great to make the wrangler logging for preview optional. I suggest we add a silence flag to `wrangler preview` that allows folks to explicitly ask for Wrangler preview without helpful log messages (I say this instead of proposing a --verbose flag because I think the `wrangler preview` default messages are quite informative to first-time Wrangler users).

**Comment by gabbifish** [comment]
Another thing we could do is make `wrangler preview --headless` also serve as our log-free version of `wrangler preview` (so it only returns the worker response, nothing else).

**Comment by ashleymichal** [comment]
the stderr/stdout problem is its own thing (new issue created in #1283 ) but the rest of this issue was solved by #816

**Comment by EverlastingBugstopper** [comment]
i dont think --headless fixed this but `wrangler dev` should let you curl an endpoint and pipe to whatever you want so im fine closing this
